arch/xtensa/core/mem_manage.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/xtensa/arch.h>
+#include <zephyr/arch/xtensa/cache.h>
+#include <zephyr/sys/arch_mem_interface.h>
+
+bool arch_mem_is_valid_phys_addr(uintptr_t phys)
+{
+	bool valid;
+	uintptr_t cached = (uintptr_t)arch_xtensa_cached_ptr((void *)phys);
+
+	valid = ((phys >= CONFIG_SRAM_BASE_ADDRESS) &&
+		 (phys < (CONFIG_SRAM_BASE_ADDRESS + (CONFIG_SRAM_SIZE * 1024UL))));
+
+	valid |= ((cached >= CONFIG_SRAM_BASE_ADDRESS) &&
+		  (cached < (CONFIG_SRAM_BASE_ADDRESS + (CONFIG_SRAM_SIZE * 1024UL))));
```

Comment:
Won't this break on discontiguous mappings?  There are holes between SRAM and IMR on intel_adsp platforms, and between SRAM and DRAM on mtk_adsp, and I'm sure other platforms have similar issues.

Seems like what we really want is a fancier API in the core kernel where you can have a sorted array of regions that these predicates search (we're at least partially there with devicetree now, I think?), rather than an arch-specific API.

Reply:
The implementation under Xtensa simply acts as default. I have marked them as __weak so SoC or board can override this if necessary. Having SoC and/or board to add to the range will get a bit complicated as it would require having to process 3 different arrays (arch, SoC and board), or that we need to add some linker magic to do iterable sections. I believe both would add too much confusion, so letting SoC or board to override the default implementation seems easier to developers.